src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -560,3 +697,58 @@ function mapAndFilterNotices(notices: GraphQlNotice[]): Notice[] {
 
   return filterNotices(mappedNotices);
 }
+
+function findTripPatternCombinationsList(
+  tripPatternCombinations: ViaTripsWithDetailsQuery['viaTrip']['tripPatternCombinations'],
+): { from: number; to: number }[][] {
+  return tripPatternCombinations.map((combinations) =>
+    combinations.map((combination) => ({
+      from: combination.from!,
+      to: combination.to!,
+    })),
+  );
+}
+
+function findTripPatternsFromViaTo(
+  tripPatternCombinations: { from: number; to: number }[][],
+  tripPatternsFromVia: ViaTripsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'],
+  tripPatternsViaTo: ViaTripsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'],
+) {
+  const tripPatterns: ViaTripsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'] =
+    [];
+  tripPatternCombinations.map((segment) => {
+    segment.map((combo) =>
+      tripPatterns.push({
+        expectedStartTime: tripPatternsFromVia[combo.from].expectedStartTime,
+        expectedEndTime: tripPatternsViaTo[combo.to].expectedEndTime,
+        legs: [
+          ...tripPatternsFromVia[combo.from].legs,
+          ...tripPatternsViaTo[combo.to].legs,
+        ],
+      }),
+    );
+  });
+  return tripPatterns;
+}
+
+function findTripPatternsFromViaToWithDetails(
+  tripPatternCombinations: { from: number; to: number }[][],
+  tripPatternsFromVia: ViaTripsWithDetailsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'],
+  tripPatternsViaTo: ViaTripsWithDetailsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'],
+) {
+  const tripPatterns: ViaTripsWithDetailsQuery['viaTrip']['tripPatternsPerSegment'][0]['tripPatterns'] =
+    [];
+  tripPatternCombinations.map((segment) => {
+    segment.map((combo) =>
+      tripPatterns.push({
+        expectedStartTime: tripPatternsFromVia[combo.from].expectedStartTime,
+        expectedEndTime: tripPatternsViaTo[combo.to].expectedEndTime,
+        legs: [
+          ...tripPatternsFromVia[combo.from].legs,
+          ...tripPatternsViaTo[combo.to].legs,
+        ],
+      }),
+    );
+  });
```

Comment:
Would these functions be a good candidate for some unit tests? I guess there could be made tests for other parts of this code as well, such as testing that the correct query is used when `via` is defined and so on.

Reply:
Yes, I should probably add a few unit tests to verify the behavior of the feature when via is defined. I´ll look into it👍